src/main/java/net/dzikoysk/funnyguilds/listener/TntProtection.java

```diff
@@ -0,0 +1,38 @@
+package net.dzikoysk.funnyguilds.listener;
+
+import net.dzikoysk.funnyguilds.FunnyGuilds;
+import net.dzikoysk.funnyguilds.basic.guild.Region;
+import net.dzikoysk.funnyguilds.basic.guild.RegionUtils;
+import net.dzikoysk.funnyguilds.data.configs.PluginConfiguration;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityExplodeEvent;
+
+import java.time.LocalTime;
+
+public class TntProtection implements Listener {
+
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+    public void onExplode(EntityExplodeEvent event) {
+        PluginConfiguration config = FunnyGuilds.getInstance().getPluginConfiguration();
```

Comment:
Or maybe we could check here instead of conditional registration of the listener if there's any of the tnt protection option enabled before doing anything?

Reply:
Okay i think, is better option.